### DISCUSSION TITLE
Active record in own form?

### DISCUSSION THREAD

**Discussion by RPotkowaVegasystems** [filamentphp/filament#8298] [Help]
I use my own form in the EditAction.

```
EditAction::make()->form(self::myCustomForm())

public static function myCustomForm()
{

...doSomeBefore...accessOnActiveRecordOrGetTheModelId

return [myformfields etc..]
}
```

My Question is, how can i access on the active Record to get the model id?

**Comment by leandrocfe** [comment]
I think you can use `$record` in the `mountUsing` method
```php
->mountUsing(function (YourModel $record, Form $form) {
    $form->fill([
        'field' => $record->field
    ]);
})
```

**Comment by RPotkowaVegasystems** [comment]
But how can i access the values inside my static formFunction()?

**Reply by leandrocfe** [reply]
Sorry, I didn't understand what you are trying to do

**Reply by RPotkowaVegasystems** [reply]
Ok, i try to describe.
I have created the form in my custom static form function, where i have some forloops etc to create different Sections/Fields etc to, create dynamic forms. 

What i try to do, is to access on the record data, to achive like: if $record->value == X, create this fields, wenn Y, create other Sections/Fields etc. with other values etc.

**Reply by leandrocfe** [reply]
Maybe you can use via $record or $livewire, like

```php
->required(fn ($record): bool => $record->value)
->required(fn ($livewire): bool => $livewire->getMountedTableActionRecord()->value)
```
